compiler/src/dotty/tools/dotc/Compiler.scala

```diff
@@ -148,6 +148,7 @@ class Compiler {
   /** Generate the output of the compilation */
   protected def backendPhases: List[List[Phase]] =
     List(new backend.sjs.GenSJSIR) :: // Generate .sjsir files for Scala.js (not enabled by default)
+    List(new GenBCodeChecks) ::       // Perform JVM specific checks
```

Comment:
That's an entire new mega phase for a couple checks. We don't want that. Either put the new miniphase in a previous group so that it can be fused with other miniphases, or integrate the checks directly into `GenBCode`.

Reply:
I agree, it just we don't have mini phases in the backend. I've moved it to the end of the transform phases (just before the backend).